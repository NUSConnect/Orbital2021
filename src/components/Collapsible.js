import React, { useState } from 'react'
import { StyleSheet, Text, TouchableOpacity, LayoutAnimation, FlatList } from 'react-native'
import CheckList from './CheckList'

export default function Collapsible ({ header, data, items, setItems, selectItem }) {
  const [open, setOpen] = useState(false)

  const  onPress = () => {
    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut)
    setOpen(!open)
  }

  return (
    <TouchableOpacity onPress={onPress} style={styles.item} testID='collapsible'>
        <Text style={open ? styles.openedText : styles.closedText}>{header}</Text>
        {open && (
          <FlatList
            style={styles.list}
            columnWrapperStyle={{ justifyContent: 'space-between' }}
            numColumns={2}
            data={data}
            keyExtractor={(item, index) => index.toString()}
            renderItem={({ item }) => (
              <CheckList
                checked={items.includes(item)}
                item={item}
                selectItem={selectItem(items, setItems)}

              />
            )}
          />
        )}
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  item: {
    width: '100%',
    borderWidth: 1,
    overflow: 'hidden',
    backgroundColor: 'orange',
    borderColor: 'white'
  },
  openedText: {
    color: 'white',
    fontSize: 24,
    paddingLeft: 10,
    paddingTop: 10
  },
  closedText: {
    color: 'white',
    fontSize: 24,
    paddingLeft: 10,
    paddingVertical: 10,
  },
  list: {
    marginTop: 20,
    width: '100%',
    backgroundColor: 'white'
  }
})
