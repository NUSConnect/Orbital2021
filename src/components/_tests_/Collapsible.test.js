import * as React from 'react'
import Collapsible from '../Collapsible'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(<Collapsible />)
  expect(tree).toMatchSnapshot()
})
