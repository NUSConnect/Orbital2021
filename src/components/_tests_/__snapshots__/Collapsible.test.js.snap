// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "orange",
      "borderColor": "white",
      "borderWidth": 1,
      "opacity": 1,
      "overflow": "hidden",
      "width": "100%",
    }
  }
  testID="collapsible"
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "paddingLeft": 10,
        "paddingVertical": 10,
      }
    }
  />
</View>
`;
